_posts/2019-06-25-lambda-quickstart.md

```diff
@@ -0,0 +1,320 @@
+        }
+    ]
+}
+```
+
+
+##### 2.2.1. Recommended setup
+- Create a [new IAM policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_create.html#access_policies_create-start), with the above minimum permissions. For example call this new policy `OpenFaaS Lambda Provider Policy`.
+- Create a [new IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user.html), and assign it the above policy. For example call this new role `OpenFaaS Lambda Provider Role`. 
+- Create a [new IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html), that will act as a service account for the `faas-lambda` provider i.e. `service-account-openfaas-lambda`. 
+**Assign** the new IAM user the `OpenFaaS Lambda Provider Role`.
+- Create [IAM access credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html) using the new service account user.
+
+#### 2.3. Create OpenFaaS Function IAM role
+When an OpenFaaS functions executes in AWS Lambda it needs an IAM role. This role will give functions access to other
+AWS services. Currently we only support one role that is assigned to all functions. In the future we will change this
+to be per function. 
+
+If your functions do not require access please follow the steps to create a [blank role](#232-no-access-function-iam-role)
+
+##### 2.3.1. Restricted function IAM Role
+Your requirements will be specific to you, in this example I am giving functions access to a particular s3 bucket:
+
+
+```javascript
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "CreateLogs",
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogGroup",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "ListObjectsInBucket",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListBucket"
+            ],
+            "Resource": [
+                "arn:aws:s3:::kubecon-2019",
+                "arn:aws:s3:::containerdays-2019"
+            ]
+        },
+        {
+            "Sid": "AllObjectActions",
+            "Effect": "Allow",
+            "Action": "s3:*Object",
+            "Resource": [
+                "arn:aws:s3:::kubecon-2019/*",
+                "arn:aws:s3:::containerdays-2019/*"
+            ]
+        }
+    ]
+}
+``` 
+
+- Create a [new IAM policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_create.html#access_policies_create-start), with the above minimum permissions. For example call this new policy `OpenFaaS Lambda Function Policy`.
+- Create a [new IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user.html), and assign it the above police. For example call this new role `OpenFaaS Lambda Function Role`. 
+
+OR
+
+##### 2.3.2. No access function IAM Role
+
```

Comment:
" No access function IAM Role"

^ what is this for?

Reply:
Good idea, I will add some extra text, it was mentioned in the previous step:

> If your functions do not require access please follow the steps to create a [blank role](#232-no-access-function-iam-role)